model/project_parser_test.go

```diff
@@ -1188,3 +1195,130 @@ tasks:
 	assert.Equal("somethingElse", proj.Loggers.Agent[1].Type)
 	assert.Equal("commandLogger", proj.Tasks[0].Commands[0].Loggers.System[0].Type)
 }
+
+func TestParserProjectPersists(t *testing.T) {
+	simpleYml := `
+loggers:
+  agent:
+    - type: something
+      splunk_token: idk
+    - type: somethingElse
+tasks:
+- name: task_1
+  depends_on:
+  - name: embedded_sdk_s3_put
+    variant: embedded-sdk-android-arm32
+  commands:
+  - command: myCommand
+    params:
+      env:
+        ${MY_KEY}: my-value
+        ${MY_NUMS}: [1,2,3]
+    loggers:
+      system:
+       - type: commandLogger
+functions:
+  run-make:
+    command: subprocess.exec
+    params:
+      working_dir: gopath/src/github.com/evergreen-ci/evergreen
+      binary: make
+      env:
+        CLIENT_URL: https://s3.amazonaws.com/mciuploads/evergreen/${task_id}/evergreen-ci/evergreen/clients/${goos}_${goarch}/evergreen
+`
+	paramsYAML := `
+  - command: myCommand
+    params:
+      env:
+        ${MY_KEY}: my-value
+        ${MY_NUMS}:
+        - 1
+        - 2
+        - 3`
+
+	for name, test := range map[string]func(t *testing.T){
+		"simpleYaml": func(t *testing.T) {
+			pp, errs := createIntermediateProject([]byte(simpleYml))
+			yamlToCompare, err := yaml.Marshal(pp)
+			assert.NoError(t, err)
+
+			assert.Len(t, errs, 0)
+			v := Version{
+				Id:            "my-version",
+				ParserProject: pp,
+				Config:        string(yamlToCompare),
+			}
+			assert.NoError(t, v.Insert())
+			newV, err := VersionFindOneId(v.Id)
+			assert.NoError(t, err)
+			require.NotNil(t, newV)
+			require.NotNil(t, newV.ParserProject)
+			require.Len(t, newV.ParserProject.Tasks, 1)
+			require.Len(t, newV.ParserProject.Tasks[0].Commands, 1)
+			assert.NotEmpty(t, newV.ParserProject.Tasks[0].Commands[0].Params)
```

Comment:
Should the test also compare `newV.ParserProject.Tasks[0].Commands[0].Params` to `newV.ParserProject.Tasks[0].Commands[0].ParamsYAML`?

Reply:
No--the reason we have ParamsYAML to begin with is because Params cannot be persisted. I am verifying that the params block appears correctly in the config which is the reason for keeping the Params field.